main.go

```diff
@@ -86,36 +86,34 @@ func readInViperConfig() error {
 		configSourceStr = overriddenConfigLocation
 	}
 
-	if configSourceStr == "" {
-		return handleConfigErr(errors.New("-cfg is required parameter. Read more about EventNative configuration: https://docs.eventnative.org/configuration-1/configuration"))
-	}
-
 	var payload *resources.ResponsePayload
 	var err error
 	if strings.HasPrefix(configSourceStr, "http://") || strings.HasPrefix(configSourceStr, "https://") {
 		payload, err = resources.LoadFromHttp(configSourceStr, "")
 	} else if strings.HasPrefix(configSourceStr, "{") && strings.HasSuffix(configSourceStr, "}") {
 		jsonContentType := resources.JsonContentType
 		payload = &resources.ResponsePayload{Content: []byte(configSourceStr), ContentType: &jsonContentType}
-	} else {
+	} else if configSourceStr != "" {
 		payload, err = resources.LoadFromFile(configSourceStr, "")
 	}
 
 	if err != nil {
 		return handleConfigErr(err)
 	}
 
-	if payload.ContentType != nil {
+	if payload != nil && payload.ContentType != nil {
 		viper.SetConfigType(string(*payload.ContentType))
 	} else {
 		//default content type
 		viper.SetConfigType("json")
 	}
 
-	err = viper.ReadConfig(bytes.NewBuffer(payload.Content))
-	if err != nil {
-		errWithContext := fmt.Errorf("Error reading/parsing viper config from %s: %v", configSourceStr, err)
-		return handleConfigErr(errWithContext)
+	if payload != nil {
+		err = viper.ReadConfig(bytes.NewBuffer(payload.Content))
+		if err != nil {
+			errWithContext := fmt.Errorf("Error reading/parsing viper config from %s: %v", configSourceStr, err)
+			return handleConfigErr(errWithContext)
```

Comment:
I would probably refrain from using the internal libs name ("viper" in this case) at all in error messages. The messages will be confusing for most people.

"Error reading/parsing config file %s: %v" would make more sense

Reply:
Fixed